osimpipeline/postprocessing.py

```diff
@@ -646,288 +667,389 @@ def plot_toeoff_pgc(gl, side, axes=None, *args, **kwargs):
 
 
-## Data Management ##
-## =============== ##
-
-def storage2numpy(storage_file, excess_header_entries=0):
-    """Returns the data from a storage file in a numpy format. Skips all lines
-    up to and including the line that says 'endheader'.
+def plot_force_plate_data(mot_file):
+    """Plots all force componenets, center of pressure components, and moment
+    components, for both legs.
 
     Parameters
     ----------
-    storage_file : str
-        Path to an OpenSim Storage (.sto) file.
-
-    Returns
-    -------
-    data : np.ndarry (or numpy structure array or something?)
-        Contains all columns from the storage file, indexable by column name.
-    excess_header_entries : int, optional
-        If the header row has more names in it than there are data columns.
-        We'll ignore this many header row entries from the end of the header
-        row. This argument allows for a hacky fix to an issue that arises from
-        Static Optimization '.sto' outputs.
+    mot_file : str
+        Name of *.mot (OpenSim Storage) file containing force plate data.
 
-    Examples
-    --------
-    Columns from the storage file can be obtained as follows:
+    """
+    data = storage2numpy(mot_file)
+    time = data['time']
+
+    pl.figure(figsize=(5 * 2, 4 * 3))
+
+    for i, prefix in enumerate([ '1_', '']):
+        pl.subplot2grid((3, 2), (0, i))
+        for comp in ['x', 'y', 'z']:
+            pl.plot(time, 1.0e-3 * data['%sground_force_v%s' % (prefix, comp)],
+                    label=comp)
+        if i == 0:
+            pl.title('left foot')
+            pl.ylabel('force components (kN)')
+        if i == 1:
+            pl.title('right foot')
+            pl.legend(loc='upper left', bbox_to_anchor=(1, 1))
+
+    for i, prefix in enumerate([ '1_', '']):
+        pl.subplot2grid((3, 2), (1, i))
+        for comp in ['x', 'y', 'z']:
+            pl.plot(time, data['%sground_force_p%s' % (prefix, comp)],
+                    label=comp)
+
+        if i == 0: pl.ylabel('center of pressure components (m)')
+
+    for i, prefix in enumerate([ '1_', '']):
+        pl.subplot2grid((3, 2), (2, i))
+        for comp in ['x', 'y', 'z']:
+            pl.plot(time, data['%sground_torque_%s' % (prefix, comp)],
+                    label=comp)
+
+        if i == 0: pl.ylabel('torque components (N-m)')
+        pl.xlabel('time (s)')
+
+
+def plot_gait_torques(output_filepath, actu, primary_leg, cycle_start,
+        cycle_end, primary_footstrike, opposite_footstrike,
+        toeoff_time=None):
+    """Plots hip, knee, and ankle torques, as a function of percent
+    gait cycle, for one gait cycle. Gait cycle starts with a footstrike (start
+    of stance). Torques are plotted for both legs; the data for the
+    'opposite' leg is properly shifted so that we plot it starting from a
+    footstrike as well.
+
+    We assume torques are in units of N-m.
+
+    Knee torque is negated: In OpenSim, a flexion torque is represented by a
```

Comment:
For my current project, I have a check where I look at the min and max value of the knee coordinate to deduce whether flexion or extension is positive. I could do something similar here. It would be nice if models had permanent "types" or something, so we could check for things like that more directly.